### ISSUE TITLE
Sub-iterators of ConjunctionDISI are not on the same document! error

### ISSUE THREAD

**Issue by bperalta** [elastic/elasticsearch#89789]
### Elasticsearch Version

8.4.1, 7.17.5

### Java Version

_bundled_

### OS Version

official docker image on Darwin MacBookBertrand.local 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:22 PDT 2022; root:xnu-8020.121.3~4/RELEASE_X86_64 x86_64

### Problem Description

Using an index with mapping containing a nested field, we get the "Sub-iterators of ConjunctionDISI are not on the same document!" error if the search use an intervals query and the searched value is empty (after stop words filtering for example).

### Steps to Reproduce

1. Create a simple index with a nested field and stop words filter
```
PUT /simple_test
{
    "settings": {
        "analysis": {
            "analyzer": {
                "default": {
                    "tokenizer": "whitespace",
                    "filter": [
                        "stop"
                    ]
                }
            }
        }
    },
    "mappings": {
        "properties": {
            "object": {
                "type": "text"
            },
            "documents": {
                "type": "nested",
                "properties": {
                    "content": {
                        "type": "text"
                    }
                }
            }
        }
    }
}
```

2. Add a document
```
POST /simple_test/_doc/1
{
    "object": "test"
}
```
No matter if the nested field has content or not

3. Execute a query with stop words
```
GET /simple_test/_search
{
    "query": {
        "intervals": {
            "object": {
                "match": {
                    "query": "not in that",
                    "max_gaps": 2,
                    "ordered": false
                }
            }
        }
    }
}
```

As the searched words are all filtered, the search result is
```
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_argument_exception",
                "reason": "Sub-iterators of ConjunctionDISI are not on the same document!"
            }
        ],
        "type": "search_phase_execution_exception",
        "reason": "all shards failed",
        "phase": "query",
        "grouped": true,
        "failed_shards": [
            {
                "shard": 0,
                "index": "simple_test",
                "node": "gstAliy6ReeeL5eSvg1IZQ",
                "reason": {
                    "type": "illegal_argument_exception",
                    "reason": "Sub-iterators of ConjunctionDISI are not on the same document!"
                }
            }
        ],
        "caused_by": {
            "type": "illegal_argument_exception",
            "reason": "Sub-iterators of ConjunctionDISI are not on the same document!",
            "caused_by": {
                "type": "illegal_argument_exception",
                "reason": "Sub-iterators of ConjunctionDISI are not on the same document!"
            }
        }
    },
    "status": 400
}
```

**Comment by dweiss** [comment]
Hi @romseygeek - I'm getting the same error in a similar scenario with Lucene 9.7.0 (well, Solr that embeds Lucene 9.7.0). Do you remember off the top of your head if this has been fixed in Lucene main? Thanks!

**Comment by dweiss** [comment]
Ok, I don't think this is fixed in Lucene - it's Intervals.analyzedText that can cause problems.

**Comment by dweiss** [comment]
Hmm... I'm confused - this happens for me with Lucene 9.7.0, yet it's fixed in 9.4.0? I'll dig tomorrow. https://github.com/apache/lucene/pull/11760

**Comment by dweiss** [comment]
False alarm - stale bug in my own code (sorry for the noise).
